### PR TITLE
Log successful transaction if we've retried

### PR DESCRIPTION
**Pull request by gsheasby**
Note that if multiple transactions are in flight simultaneously, it
won't be clear *which* transaction completed after the retry.

To do:

- [x] ensure the logging / logic is shared between this and the other txMgr retry loop (AbstractLockAwareTransactionManager#runTaskWithLocksWithRetry)

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1376)

### CODE REVIEW

File: atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractTransactionManager.java [hunk 1]
```diff
@@ -49,7 +53,7 @@
                     log.warn("Failing after {} tries.", failureCount, e);
                     throw Throwables.rewrap(String.format("Failing after %d tries.", failureCount), e);
                 }
-                log.info("retrying transaction", e);
+                log.info("Retrying transaction.", e);
```
Comment (hsaraogi) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractTransactionManager.java]: ``log.info("Retrying transaction after {} failure(s)",  failureCount);``

File: atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractTransactionManager.java [hunk 2]
```diff
@@ -38,7 +38,11 @@
         while (true) {
             checkOpen();
             try {
-                return runTaskThrowOnConflict(task);
+                T result = runTaskThrowOnConflict(task);
+                if (failureCount > 0) {
+                    log.info("Successfully completed transaction after {} retries.", failureCount);
```
Comment (schlosna) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractTransactionManager.java]: We should generate a unique transaction ID (say just a UUID but separate from its logical transaction timestamp) that we log as part of retry. 

Would also be useful to expose overall transaction manager metrics for retries (including rate of retry and time spent in retries), but that can happen later after #1356

Comment (jboreiko) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractTransactionManager.java]: This is somewhat covered by the traceId's being logged out.  That said, a particular span could be running multiple transactions, but at least the success should follow a single failure for a particular traceId in single threaded AtlasDB client work flows.

**Comment by jboreiko** [comment]
So we can tack a UUID into the retry loop to get us unique identifiers for this particular retry run.  We should also ensure the logging / logic is shared between this and the other txMgr retry loop (AbstractLockAwareTransactionManager#runTaskWithLocksWithRetry)

**Comment by gsheasby** [comment]
Thanks for the addition, @jboreiko - this is ready for review again.

**Pull request merged**